include/libtorrent/bdecode.hpp

```diff
@@ -202,6 +202,9 @@ struct bdecode_token
 
 	int start_offset() const { TORRENT_ASSERT(type == string); return int(header) + 2; }
 
+	// diff between current and next item offset
+	int diff_size() const { bdecode_token const* next(this); ++next; return next->offset - offset; }
```

Comment:
this is a very dangerous function. it makes assumptions about where its object is allocated, or rather, that another identical object is allocated immediately following it. For every array, this assumption is false for one element.

I don't think this belongs as a member function, as it cannot satisfy the basic assumption that the following code would be valid:

```
bdecode_token a;
a.diff_size();
```